docs/source/frameworks/onnx.rst

```diff
@@ -0,0 +1,341 @@
+      # ONNX Runtime will return a list of outputs
+      ort_outs = ort_session.run(None, ort_inputs)
+      print(ort_outs[0])
+
+.. tab:: TensorFlow
+
+   First let's install ``tf2onnx``
+
+   .. code-block:: bash
+
+      pip install tf2onnx
+
+.. tab:: Scikit Learn
+
+   TODO
+
+
+Saving ONNX model with BentoML
+-----------------------------
+
+To quickly save an ONNX model to BentoML's :ref:`Model
+Store<concepts/model:Managing Models>`, first use ``onnx.load`` to
+load the exported ONNX model back into ``onnx.ModelProto`` object,
+then call BentoML's ``save_model``:
+
+
+.. code-block:: python
+
+   signatures = {
+       "run": {"batchable": True},
+   }
+   bentoml.onnx.save_model("super_resolution", onnx_model, signatures=signatures)
+
+which will result:
+
+.. code-block:: bash
+
+   Model(tag="super_resolution:lwqr7ah5ocv3rea3", path="~/bentoml/models/super_resolution/lwqr7ah5ocv3rea3/")
+
+.. note::
+
+   ``save_model`` will use ``{"run": {"batchable": False}}`` as
+   default signatures if ``signatures`` is not provided. Set
+   ``batchable`` to ``False`` will disable BentoML's
+   :ref:`guides/batching:Adaptive Batching` functionality. That's why
+   we provide our own signatures here. Read more about :ref:`Model
+   Signatures <concepts/model:Model Signatures>` and :ref:`Batch Input
+   <concepts/model:Batching>`
+
+.. seealso::
+
+   ``save_model`` also has some :ref:`general options
+   <concepts/model:Save A Trained Model>` for functionalities like
+   saving metadata and custom objects.
+
+
+Building a Service for **ONNX**
+-------------------------------
+
+.. seealso::
+
+   :ref:`Building a Service <concepts/service:Service and APIs>` for how to
+   create a prediction service with BentoML.
+
+.. code-block:: python
+
+   import bentoml
+
+   import numpy as np
+   from PIL import Image as PIL_Image
+   from PIL import ImageOps
+   from bentoml.io import Image
+
+   runner = bentoml.onnx.get("super_resolution:latest").to_runner()
+
+   svc = bentoml.Service("super_resolution", runners=[runner])
+
+   @svc.api(input=Image(), output=Image())
+   def sr(img) -> np.ndarray:
+       img = img.resize((224, 224))
+       gray_img = ImageOps.grayscale(img)
+       arr = np.array(gray_img) / 255.0 # convert from 0-255 range to 0.0-1.0 range
+       arr = np.expand_dims(arr, (0, 1)) # add batch_size, color_channel dims
+       sr_arr = runner.run.run(arr)
+       sr_arr = np.squeeze(sr_arr) # remove batch_size, color_channel dims
+       sr_img = PIL_Image.fromarray(np.uint8(sr_arr * 255) , 'L')
+       return sr_img
+
+.. note::
+
+   In above codes we use ``runner.run.run(input_data)`` to do
+   inference. The first ``run`` is referring to
+   ``onnxruntime.InferenceSession``'s ``run`` method, while the second
+   ``run`` is BentoML's naming convention for doing runner inference
+   for a model method. For example, for a Keras model with ``predict``
+   method, we will call ``runner.predict.run(input_data)``.
```

Comment:
@aarnphm Do you think this explains `runner.run.run` clearly enough?

Reply:
```suggestion
.. note::

   In above example, notice that there are two :code:`run` in ``runner.run.run(input_data)`` while running inference. The distinction between the two ``run`` are as follow:
   1.  The first ``run`` refers  to``onnxruntime.InferenceSession``'s ``run`` method, which is the API from onnxruntime to run `inference <https://onnxruntime.ai/docs/api/python/api_summary.html#data-inputs-and-outputs>`_.
   2. The second ``run`` refers to BentoML's runner inference API for invoking a model's signature. In the case of ONNX, it happens to have the same name as the ``InferenceSession`` endpoint.
```
@ssheng does this make sense?

Reply:
Aaron's modification looks more clear than mine so I will commit this suggestion (though I still think it's better to keep the keras's example because runner.predict.run() will give user a comparison). cc @ssheng @aarnphm

Reply:
I think the keras distinction example should be left at a runners level docs. Let me draft a docs that talk dissect this components in a different docs pr.

Reply:
I think having another example would be helpful. I'm slowly leaning towards including more information in the framework guides than less.